watertap/unit_models/tests/test_thickener_unit.py

```diff
@@ -429,3 +441,403 @@ def test_list_error():
+    m.fs.unit.diameter.fix()
+
+    solver = get_solver()
+    solver.solve(m)
+
+    return m
+
+
+def scale_vars_with_scalers(m):
+    scaler = ThickenerScaler()
+    scaler.scale_model(
+        m.fs.unit,
+        submodel_scalers={
+            m.fs.unit.mixed_state: ASM1PropertiesScaler,
+            m.fs.unit.underflow_state: ASM1PropertiesScaler,
+            m.fs.unit.overflow_state: ASM1PropertiesScaler,
+        },
+    )
+
+
+def scale_vars_with_iscale(m):
+    # Set scaling factors for badly scaled variables
+    iscale.set_scaling_factor(m.fs.unit.underflow_state[0.0].flow_vol, 1e4)
+    iscale.set_scaling_factor(m.fs.unit.underflow_state[0.0].pressure, 1e-6)
+    iscale.set_scaling_factor(m.fs.unit.underflow_state[0.0].conc_mass_comp["S_S"], 1e4)
+    iscale.set_scaling_factor(
+        m.fs.unit.underflow_state[0.0].conc_mass_comp["S_NH"], 1e4
+    )
+    iscale.set_scaling_factor(
+        m.fs.unit.underflow_state[0.0].conc_mass_comp["S_ND"], 1e4
+    )
+    iscale.set_scaling_factor(m.fs.unit.overflow_state[0.0].pressure, 1e-6)
+    iscale.set_scaling_factor(m.fs.unit.overflow_state[0.0].conc_mass_comp["S_S"], 1e4)
+    iscale.set_scaling_factor(m.fs.unit.overflow_state[0.0].conc_mass_comp["S_NH"], 1e4)
+    iscale.set_scaling_factor(m.fs.unit.overflow_state[0.0].conc_mass_comp["S_ND"], 1e4)
+    iscale.set_scaling_factor(m.fs.unit.overflow_state[0.0].conc_mass_comp["X_ND"], 1e4)
+
+    iscale.calculate_scaling_factors(m.fs.unit)
+
+
+def perturb_solution(m):
+    m.fs.unit.inlet.flow_vol.fix(300 * 0.8 * units.m**3 / units.day)
+    m.fs.unit.inlet.conc_mass_comp[0, "S_I"].fix(
+        28.0643 * 0.55 * units.mg / units.liter
+    )
+
+
+@pytest.mark.requires_idaes_solver
+@pytest.mark.unit
+def test_scaling_profiler_with_scalers():
+    sp = ScalingProfiler(
+        build_model=build_model,
+        user_scaling=scale_vars_with_scalers,
+        perturb_state=perturb_solution,
+    )
+
+    stream = StringIO()
+
+    sp.report_scaling_profiles(stream=stream)
+
+    expected = """
+============================================================================
+Scaling Profile Report
+----------------------------------------------------------------------------
+Scaling Method           || User Scaling           || Perfect Scaling
+Unscaled                 || 1.384E+07 | Solved 1   ||
+Vars Only                || 7.863E+06 | Solved 1   || 2.767E+12 | Solved 1  
+Harmonic                 || 7.863E+06 | Solved 1   || 1.083E+03 | Solved 1  
+Inverse Sum              || 7.863E+06 | Solved 1   || 1.968E+03 | Solved 1  
+Inverse Root Sum Squares || 7.863E+06 | Solved 1   || 1.950E+03 | Solved 1  
+Inverse Maximum          || 7.863E+06 | Solved 1   || 1.968E+03 | Solved 1  
+Inverse Minimum          || 7.863E+06 | Solved 1   || 1.443E+03 | Solved 1  
+Nominal L1 Norm          || 7.863E+06 | Solved 1   || 4.416E+02 | Solved 1  
+Nominal L2 Norm          || 7.863E+06 | Solved 1   || 4.802E+02 | Solved 1  
```

Comment:
Interest pattern once again (like noted in #1543)--going forward, a good strategy could be to run this scaling profile report and construct the Scaler class around the report (e.g., perfect scaling, or close to it, combined with the constraint scaling method that leads to lowest condition number). Just to keep in mind